### DISCUSSION TITLE
Symfony couldn't autowire Response into controller method

### DISCUSSION THREAD

**Discussion by TomLepsky** [symfony/symfony#49085] [Q&A]
Trying to inject Symfony\Component\HttpFoundation\Response into controller method

```
namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\HttpKernel\Attribute\AsController;
use Symfony\Component\Routing\Annotation\Route;

#[AsController]
class TestController extends AbstractController
{
    #[Route(path: 'test', name: 'test', methods: ['GET'])]
    public function test(Request $request, Response $response) : Response
    {
        return $response->setContent("Hello!");
    }
}
````

But get an error
```
Could not resolve argument $response of &quot;App\Controller\TestController::test()&quot;, maybe you forgot
        to register the controller as a service or missed tagging it with the controller.service_arguments;?
        (500 Internal Server Error)
```

Debug `php bin/console debug:autowiring --all` shows, that container have `Response`

```
...
 Request represents an HTTP request.
 Symfony\Component\HttpFoundation\Request

 Response represents an HTTP response.
 Symfony\Component\HttpFoundation\Response
...
```
So, Request injected, but response no
Please tell me, why some classes couldn't be injected, even if they are present in the contsiner?

**Comment by MatTheCat** [comment]
Which version of Symfony are you using? https://github.com/symfony/symfony/pull/46279 should exclude `Response` from the container, and https://github.com/symfony/symfony/pull/48172 should prevent autowiring.

Also why are you trying to inject a response? Where is it supposed to come from?

**Reply by TomLepsky** [reply]
symfony version 6.2.
I just want to understand, why `Response` present in container, but it couldn't be injected

**Reply by MatTheCat** [reply]
Because it is not a service (read https://github.com/symfony/symfony/pull/46279).
